reframe/frontend/statistics.py

```diff
@@ -70,46 +74,103 @@ def retry_report(self):
 
         return '\n'.join(report)
 
+    @property
+    def report_dict(self):
+        _report = []
+        current_run = rt.runtime().current_run
+        for tl, tr in zip(self._tasks, range(current_run + 1)):
+            for t in tl:
```

Comment:
First, I suggest renaming `self._tasks` to `self._alltasks` and then rewrite the loop as follows:

```suggestion
        for run_no, run in enumerate(self._alltasks):
            for task in run:
```